### PR TITLE
Add associations to School library OOP project

### PR DESCRIPTION
**Pull request by ismayito**
### Added associations from UML diagram
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------#### Summary:
_______________________________________________________________________________________________________
In this I added the following
- Created a class Classroom and instantiated it
- Created a relationship between classes Classroom and Student
- Created a classes Book and Rental then instantiated them
- Added many-to-many relationship between class Person and Book using Rental as intermediate class

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(book_rented)
+    @rentals.push(book_rented)
+    book_rented.book = self
+  end
```
Comment (codepantha) [inline book.rb]: - [ ] In the `initialize` method of the `Rental` class, we can see that the `rental` is adding itself to the `@book` achieving the 'belongs-to' association. Therefore, we only need to make sure that the book itself is passed in as a parameter to create the new rental. Please modify your code like this 👇 
```ruby
def add_rental(person, date)
  Rental.new(date, person, self)
end
```

File: person.rb [hunk 2]
```diff
@@ -1,20 +1,26 @@
...
-  attr_reader :id
-
   def correct_name
     @name
   end
 
+  def add_rentals(rental)
+    @rentals << rental
+    rental.person = self
+  end
```
Comment (codepantha) [inline person.rb]: - [ ] In the `initialize` method of the `Rental` class, we can see that the `rental` is adding itself to the `@person` achieving the 'belongs-to' association. Therefore, we only need to make sure that the person itself is passed in as a parameter to create the new rental. Please modify your code like this 👇 
```ruby
def add_rental(book, date)
  Rental.new(date, self, book)
end
```

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @ismayito 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Created associations between classes 📌 
✔️ No linter error 🟢 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+require './rental'
```
Comment (OyePriscilla) [inline book.rb]: - [ ] Please kindly note that your app did not function as it was in the previous review due to the require './rental' method that you newly added. **Note: You can remove it for the app to function as it was**.
 I used this snippet that you implemented to check your app as well. Thanks.
```
person = Person.new(22, 'mayito')
book = Book.new('comic', 'author')
rent = Rental.new('12-02-2023', person, book)
p rent
```
Untitled

File: person.rb [hunk 4]
```diff
@@ -1,20 +1,26 @@
 require './decorator'
+require './rental'
```
Comment (OyePriscilla) [inline person.rb]: - [ ] Please kindly note that your app did not function as it was in the previous review due to the require './rental' method that you newly added. **Note: You can remove it for the app to funtion well**.

 I used this snippet that you implemented to check your app as well. Thanks.
```
person = Person.new(22, 'mayito')
book = Book.new('comic', 'author')
rent = Rental.new('12-02-2023', person, book)
p rent
```
Untitled

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @ismayito 👏 👏 👏, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

so-close-so-close-seth-meyers

 ## Highlights
Passed appropriate parameter to create rental ✔️ 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi @ismayito 👋, 

Great job implementing the required changes by the previous reviewer 👍 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



✔️ Descriptive PR
✔️ Created associations between classes 📌
✔️ No linter error 🟢

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**